e2e/test_operator_upgrades_variations/operator_upgrades_variations_test.go

```diff
@@ -79,11 +79,11 @@ func clusterSetupWithConfig(config testConfig) *fixtures.FdbCluster {
 	return cluster
 }
 
-func performUpgrade(config testConfig, validateFunc func(cluster *fixtures.FdbCluster)) {
+func performUpgrade(config testConfig, preUpgradeFunction func(cluster *fixtures.FdbCluster)) {
 	fdbCluster = clusterSetupWithConfig(config)
 	startTime := time.Now()
+	preUpgradeFunction(fdbCluster)
```

Comment:
Looking at the usages, I am a bit confused as to how this passes? the callers seem to use it as a validation function

Reply:
That's correct, in most test cases this was used as a pre-validation step. The validation should happen before the upgrade is triggered to make sure the conditions are met before spending time on the upgrade.

Reply:
ah, I was understanding cases where cluster config is set e.g. https://github.com/johscheuer/fdb-kubernetes-operator/blob/6a1ea6c10d5750ebff7d040a4c1d9c9f19cc6cb9/e2e/test_operator_upgrades_variations/operator_upgrades_variations_test.go#L197 as a new change instead of version change, that makes more sense